### PR TITLE
cleaner way to launch the profiling assertions

### PR DESCRIPTION
**Pull request by agallou**
assertions could now be run like this :

``` php
            $this
                ->blackfire($this->getBlackfireClient())
                    ->assert('main.wall_time < 2s', "Temps d'execution")
                    ->profile(function() {
                        sleep(4);
                        //some code of atoum assertions
                        $this->boolean(true)->isTrue();
                    })
            ;
```

There is no more need to instanciate a configuration object.

### CODE REVIEW

**Comment by Hywan** [comment]
:+1:!

File: README.md [hunk 1]
```diff
@@ -17,39 +17,32 @@ use atoum;
...
-            $this->boolean(true)->isTrue();
-        };
-
         $this
-            ->blackfireProfile($this->getBlackfireClient(), $callback, $config)
-            ->matchesAssertions()
+            ->blackfire($this->getBlackfireClient())
+                ->assert('main.wall_time < 2s', "Temps d'execution")
+                ->profile(function() {
+                    sleep(4);
```
Comment (Hywan) [inline README.md]: Why a sleep?

File: classes/asserters/blackfire.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+     *
+     * @return \mageekguy\atoum\asserters\object
+     */
+    public function profile($callback)
+    {
+        $client = $this->getClient();
+
+        if (!($callback instanceof \closure))
+        {
+            $this->fail($this->_('profile parameter is not a closure'));
```
Comment (Hywan) [inline classes/asserters/blackfire.php]: We receive a **parameter** but the user passes an **argument**, so the fail message should be: `Profile argument is not a closure.` I guess. Thoughts?

File: README.md [hunk 3]
```diff
@@ -17,39 +17,32 @@ use atoum;
...
+                    //some code of atoum assertions
+                    $this->boolean(true)->isTrue();
+                })
```
Comment (fabpot) [inline README.md:36]: It is indeed nicer, but only proxies the `assert()` method from `Configuration`. Defining custom metrics for instance is very common in such unit tests. In this case, you would need to fallback to using a proper configuration object. What about proxying all methods from `Configuration`? Or just add a `__call()` method to automatically proxy them?

Comment (Hywan) [inline README.md:36]: I guess it's fair enough to start with a `__call` to proxy everything :+1:. If we need deeper instrumentation, we could override methods one by one after.

File: README.md (hunk 1 above)
Comment (Hywan) [inline README.md]: If the goal is to wait the data being sent, we must use a promise here.

Comment (fabpot) [inline README.md]: I think it's just to make the assertion fail (4s > 2s)

Comment (Hywan) [inline README.md]: Oh, yup!
@agallou Maybe a comment would be welcomed :-).

File: README.md [hunk 4]
```diff
@@ -17,39 +17,32 @@ use atoum;
...
-
         $this
-            ->blackfireProfile($this->getBlackfireClient(), $callback, $config)
-            ->matchesAssertions()
+            ->blackfire($this->getBlackfireClient())
+                ->assert('main.wall_time < 2s', "Temps d'execution")
+                ->profile(function() {
+                    sleep(4);
+                    //some code of atoum assertions
+                    $this->boolean(true)->isTrue();
```
Comment (jubianchi) [inline README.md:35]: I think we should add a warning here when using atoum assertions in the callback: all the atoum logic will also be profiled and add its time to the global profile.

File: classes/asserters/blackfire.php (hunk 2 above)
Comment (jubianchi) [inline classes/asserters/blackfire.php]: why not just use a typehint here ?

Comment (jubianchi) [inline classes/asserters/blackfire.php]: or if you want to stick to a check here (with the `if`)  please allow any callable (using `is_callable`)

File: README.md (hunk 3 above)
Comment (agallou) [inline README.md:36]: there is a setConfiguration method on the asserter,  to use a custom \Blackfire\Profile\Configuration but it's not "easy" to use.
i will add the  __call to a list of the Configuration methods, to ease defining those custom metrics.

**Pull request merged**